src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertPlaceholderToInterpolatedStringRefactoringProvider.cs

```diff
@@ -270,7 +270,7 @@ private static TArgumentSyntax GetArgument(SeparatedSyntaxList<TArgumentSyntax>
             {
                 var argumentExpression = syntaxFacts.GetExpressionOfArgument(GetArgument(arguments, i, syntaxFacts));
                 var convertedType = semanticModel.GetTypeInfo(argumentExpression).ConvertedType;
-                if (convertedType == null)
+                if (convertedType is null || convertedType.SpecialType is SpecialType.System_Object)
```

Comment:
i'm not a fan of this here.  the existing code seems correct.  it feels like we may have a bug in the remove-unnecessary-cast code to remove this though.  I can point you to that if interested!

Reply:
>  it feels like we may have a bug in the remove-unnecessary-cast code

I don't think so. You see, this codefix attempted to add an `(object)` cast in both nullable and non-nullable scenarios. After the codefix there is a code reducer pass, and it initially removed this cast only in non-nullable context. The reason is because in nullable context this cast carryes additional piece of information about nullability of the type. If we "fix" this in remove-unnecessary-cast part of the codebase, it can (and probably will) result in bugs, where applying some codefix results in nullability warning.

Reply:
>  If we "fix" this in remove-unnecessary-cast part of the codebase, it can (and probably will) result in bugs, where applying some codefix results in nullability warning.

I'm failing to see how, in the case of interpolated strings.  Especially for normal interpolations unrelated to interpolation-handlers.

This is not a general concern about removing `(object)` from all locations.  It's a specific concern that it should always be safe to remove `(object)` in an interpolation in a normal interpolated string, regardless of nullability concerns.  Indeed, if hte user had explicitly written that, we'd want to offer to simplify appropriately.

Reply:
What if we add `(object?)` instead of `(object)` in a nullable context? This should always be valid, and should always be removed by the simplifier.

Reply:
@sharwell What if the user has wrote the `(object)` cast himself, shouldn't the "remove unnecessary cast" analyzer report a diagnostic? If it doesn't currently, then I'm more with @CyrusNajmabadi approach which will fix two issues: the referenced issue and a false negative bug of "remove unnecessary cast". So I think `CastSimplifier.IsUnnecessaryCast` should be corrected for this case.